packages/salesforcedx-vscode-core/src/conflict/componentDiffer.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { SourceComponent } from '@salesforce/source-deploy-retrieve/lib/src/resolve';
+import * as fs from 'fs';
+import * as path from 'path';
+
+export interface ComponentDiff {
+  projectPath: string;
+  cachePath: string;
+}
+
+export class ComponentDiffer {
```

Comment:
Do these methods need to be part of a class? Could they be standalone functions too?

Reply:
My instinct is always to have a class, but there ended up being less methods than I first anticipated, so they could be standalone static methods. Would you put these methods in their own file still or include them in TimestampConflictDetector?

Reply:
Ah that makes sense! We could keep these methods in this file and just change the signatures to be `export diffComponents()` and keep the other method a private utility function

Reply:
My recent changes are to implement this! I ended up reworking the unit test a lot, where now it writes and reads from temporary files